### PR TITLE
feat: sonobuoy collector

### PR DESCRIPTION
**Pull request by emosbaugh**
## Description, Motivation and Context

Retrieve sonobuoy results from the cluster and store them in the bundle. An analyzer would be beneficial as a follow up.

https://github.com/replicatedhq/troubleshoot.sh/pull/547

## Checklist

- [x] New and existing tests pass locally with introduced changes.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) are informative and highlight any breaking changes
- [x] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: pkg/collect/sonobuoy_results.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+// sonobuoyRetrieveResults copies results from a sonobuoy run into a Reader in tar format.
+// It also returns a channel of errors, where any errors encountered when writing results
+// will be sent, and an error in the case where the config validation fails.
+func sonobuoyRetrieveResults(
+	ctx context.Context, client kubernetes.Interface, restConfig *rest.Config, namespace, podName, containerName, path string,
+) (io.Reader, <-chan error, error) {
+	ec := make(chan error, 1)
+
+	restClient := client.CoreV1().RESTClient()
+	req := restClient.Post().
```
Comment (banjoh) [inline pkg/collect/sonobuoy_results.go:137]: Nit: Could you log the command being run? When testing this I couldn't get results so wanted to test using `kubectl`. Grabbing the parameters from logs would be nice

Some high level logs like "results were collected", "could not collect anything... "...

**Review by banjoh — APPROVED** [review]

**Review by banjoh — COMMENTED** [review]
Hmm. I'm not able to collect any results using `k exec -n sonobuoy -it sonobuoy -c kube-sonobuoy -- /sonobuoy splat /tmp/sonobuoy` or when running `support-bundle`. I'm probably missing something here. How can I test this?

**Review by banjoh — DISMISSED** [review]
Comment to remove approve.

**Comment by emosbaugh** [comment]
> Hmm. I'm not able to collect any results using `k exec -n sonobuoy -it sonobuoy -c kube-sonobuoy -- /sonobuoy splat /tmp/sonobuoy` or when running `support-bundle`. I'm probably missing something here. How can I test this?

You will first have to run sonobuoy in your cluster

```
sonobuoy run --mode quick --wait
```

File: pkg/collect/sonobuoy_results.go (hunk 1 above)
Comment (emosbaugh) [reply pkg/collect/sonobuoy_results.go:137]: Added

```
 Collecting support bundle ⠙ sonobuoyI0215 06:46:55.370095   50799 sonobuoy_results.go:132] Sonobuoy collector runing command: kubectl exec -n sonobuoy sonobuoy -c kube-sonobuoy -- /sonobuoy splat /tmp/sonobuoy
 Collecting support bundle ⠼ sonobuoyI0215 06:46:56.623800   50799 sonobuoy_results.go:99] Sonobuoy collector found file: 202402151442_sonobuoy_ca07019f-7be9-4c54-9449-e1611c7cc5a8.tar.gz
```

**Review by banjoh — APPROVED** [review]

**Pull request merged**